scripts/build_go.sh

```diff
@@ -142,6 +142,25 @@ if [[ "$agpl" == 1 ]]; then
 	# a flag to control AGPL vs. enterprise behavior.
 	ldflags+=(-X "'github.com/coder/coder/v2/buildinfo.agpl=true'")
 fi
+cgo=0
+if [[ "$dylib" == 1 ]]; then
+	if [[ "$os" != "darwin" ]]; then
+		error "dylib builds are not supported on $os"
+	fi
+	cgo=1
+	build_args+=("-buildmode=c-shared")
+	SDKROOT="$(xcrun --sdk macosx --show-sdk-path)"
+	export SDKROOT
+	bin_ident="com.coder.vpn"
```

Comment:
the desktop is `com.coder.Coder-Desktop` and the network extension is `com.coder.Coder-Desktop.VPN` --- so maybe this should be `com.coder.Coder-Desktop.VPN.dylib`